client/packages/programs/src/JsonForms/common/components/Array/Notes.tsx

```diff
@@ -0,0 +1,279 @@
+    })
+    .optional(),
+});
+
+const NotesComponent = (props: ArrayControlCustomProps) => {
+  const { enabled, data, config } = props;
+  const { localisedDateTime } = useFormatDateTime();
+
+  const options = props.uischema.options;
+
+  const inputData = data ?? [];
+
+  // This injects the required details info so that the children of "Notes" will
+  // be "Note" components without having to specify it specifically in the
+  // UISchema (since that would be redundant)
+  if (!props.uischema.options) props.uischema.options = {};
+  if (!props.uischema?.options?.['detail']) {
+    props.uischema.options['detail'] = {
+      type: 'VerticalLayout',
+      elements: [
+        {
+          type: 'Note',
+          scope: '/properties',
+        },
+      ],
+    };
+  }
+
+  const getItemLabel = (_: JsonData, index: number, isExpanded: boolean) => {
+    const {
+      text = '',
+      created,
+      authorName,
+    } = (inputData ? inputData[index] : {}) as {
+      text?: string;
+      created?: string;
+      authorName?: string;
+    };
+
+    return (
+      <div>
+        {!isExpanded ? (
+          <Typography
+            sx={{
+              textAlign: 'right',
+              whiteSpace: 'nowrap',
+              overflow: 'hidden',
+              textOverflow: 'ellipsis',
+              maxWidth: 350,
+            }}
+          >
+            {inputData && text}
+            <br />
+            <Typography
+              component="span"
+              sx={{
+                textAlign: 'right',
+                fontSize: '90%',
+                color: 'gray.dark',
+              }}
+            >
+              {created && `${authorName} (${localisedDateTime(created)})`}
+            </Typography>
+          </Typography>
+        ) : null}
+      </div>
+    );
+  };
+
+  const isElementEditable = (child: any, index: number) => {
+    if (!enabled) return false;
+    if (!options?.['editRestrictions']) return true;
+
+    const restrictions = options?.['editRestrictions'];
+
+    // Must be editable if no timestamp (means it's just created)
+    if (!child.created) return true;
+
+    // Only allow the latest element to be edited
+    if (restrictions?.latest && index !== inputData.length - 1) return false;
+
+    // Author must match the current user
+    if (
+      restrictions?.isCurrentUser &&
+      child?.authorId &&
+      child.authorId !== config.user.id
+    )
+      return false;
+
+    // Must not be older than `maxAge` days
+    if (DateUtils.ageInDays(child.created) >= 1) return false;
+
+    return true;
+  };
+
+  return (
+    <ArrayCommonComponent
+      {...props}
+      isElementEditable={isElementEditable}
+      getItemLabel={getItemLabel}
+      zOptions={NotesOptions}
+    />
+  );
+};
+
+export const NotesControl = withJsonFormsArrayControlProps(
+  NotesComponent as ComponentType<ArrayControlProps>
+);
+
+// NOTE component -- the child elements of the above NOTES array wrapper
+
+const NoteOptions = z
```

Comment:
Moved this into the same file since we want "Note" to be tightly coupled to the "Notes" wrapper (they should only ever be used together)

Reply:
No actual changes in the "Note" component though, except to use the "debounce" hook

Reply:
How can you configure the Note component isn't it only use internally?

Reply:
Yeah, good point. I'll need to move the Note options into the "Notes" component.

Reply:
Done -- Note options passed down by mutating the "options" in the schema, as per the renderer above.